metaflow/plugins/aws/step_functions/step_functions.py

```diff
@@ -307,19 +310,20 @@ def _visit(node, workflow, exit_node=None):
                 workflow.add_state(state.next(cardinality_state_name))
                 cardinality_state = (
                     State(cardinality_state_name)
-                    .dynamo_db(SFN_DYNAMO_DB_TABLE, "$.JobId", "for_each_cardinality")
+                    .dynamo_db(SFN_DYNAMO_DB_TABLE, "$.JobId", "for_each_cardinality, root_run_id")
                     .result_path("$.Result")
                 )
                 iterator_name = "*%s" % node.out_funcs[0]
                 workflow.add_state(cardinality_state.next(iterator_name))
                 workflow.add_state(
-                    Map(iterator_name)
+                    Map(iterator_name, self.use_distributed_map)
                     .items_path("$.Result.Item.for_each_cardinality.NS")
                     .parameter("JobId.$", "$.JobId")
                     .parameter("SplitParentTaskId.$", "$.JobId")
                     .parameter("Parameters.$", "$.Parameters")
                     .parameter("Index.$", "$$.Map.Item.Value")
-                    .next(node.matching_join)
+                    .parameter("RootRunId.$", "$.Result.Item.root_run_id.S")
+                    .next(f"{iterator_name}_GetManifest" if self.use_distributed_map else node.matching_join)
```

Comment:
If we are using the distributed map, the next step won't be the `node.matching_join`, but some additional steps to get the parameters from one of the distributed map's S3 data. To achieve that we add:

* Write the data to S3 using `ResultWriter`
* Add a `S3.GetObject` step to get the `manifest.json` file which gives us the list of files our results are saved in
* Add a `distributed map` step (with a limit of 1 value) to read just the first entry of the first file
* Add a `pass` step to just transform the data to the expected format to be passed to the join